src/legacy/core_plugins/console/public/src/mappings.js

```diff
@@ -256,61 +256,64 @@ function clear() {
   templates = [];
 }
 
-function retrieveAutocompleteInfoFromServer() {
+function getPromise(settingsKey, changedFields) {
   const autocompleteSettings = settings.getAutocomplete();
-  let mappingPromise;
-  let aliasesPromise;
-  let templatesPromise;
-  if (autocompleteSettings.fields) {
-    mappingPromise = es.send('GET', '_mapping', null, null, true);
-  }
-  else {
-    mappingPromise = new $.Deferred();
-    mappingPromise.resolve();
-  }
-  if (autocompleteSettings.indices) {
-    aliasesPromise = es.send('GET', '_aliases', null, null, true);
-  }
-  else {
-    aliasesPromise = new $.Deferred();
-    aliasesPromise.resolve();
-  }
-  if (autocompleteSettings.templates) {
-    templatesPromise = es.send('GET', '_template', null, null, true);
+  const settingsMetadata = {
+    fields: { path: '_mapping' },
+    indices: { path: '_aliases' },
+    templates: { path: '_template' },
+  };
+  let settingsPromise;
+  changedFields = changedFields ? changedFields : autocompleteSettings;
+  // Fetch autocomplete info if setting is set to true, and if user has made changes
+  if (autocompleteSettings[settingsKey] && changedFields[settingsKey]) {
+    settingsPromise = es.send('GET', settingsMetadata[settingsKey].path, null, null, true);
   } else {
-    templatesPromise = new $.Deferred();
-    templatesPromise.resolve();
+    settingsPromise = new $.Deferred();
+    // If a user has saved settings, but a field remains true, no need to make changes
+    if (autocompleteSettings[settingsKey]) {
+      settingsPromise.resolveWith(this, [{ unchanged: true }]);
+    } else {
+      settingsPromise.resolveWith(this, [ [JSON.stringify({})] ]);
```

Comment:
the previous code called `resolve()` if the setting was `false`. for example:

```
aliasesPromise = new $.Deferred();
aliasesPromise.resolve();
```

then in the callback:
```
if (aliases) {
  loadAliases(JSON.parse(aliases[0]));
} else {
  aliases = [{}];
  loadAliases({});
}
```

thought it would be cleaner to just resolve with this value instead. let me know if you think there's a better way to handle this.

Reply:
Ah OK! I think this is good, but maybe we could add a comment to clarify.

```js
// If the user doesn't want autocomplete suggestions, then clear any that exist.
```